File: template.cpp
#include <bits/stdc++.h>
using namespace std;

// =================== Макросы и типы ===================
#define int long long
#define pii pair<int, int>
#define vi vector<int>
#define vvi vector<vi>
#define all(x) (x).begin(), (x).end()
#define sz(x) ((int)(x).size())
#define pb push_back
#define ff first
#define ss second

void fast_io() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
}

// =================== Математика ===================
int gcd(int a, int b) { return b ? gcd(b, a % b) : a; }
int lcm(int a, int b) { return a / gcd(a, b) * b; }

int binpow(int a, int b, int m) {
    int res = 1;
    a %= m;
    while (b) {
        if (b & 1) res = res * a % m;
        a = a * a % m;
        b >>= 1;
    }
    return res;
}

// =================== Бинарный поиск ===================
int bin_search(int l, int r, function<bool(int)> check) {
    while (l < r) {
        int m = (l + r) / 2;
        if (check(m)) r = m;
        else l = m + 1;
    }
    return l;
}

// =================== Coordinate Compression ===================
vi compress(vi& a) {
    vi b = a;
    sort(all(b));
    b.erase(unique(all(b)), b.end());
    vi res;
    for (int x : a)
        res.pb(lower_bound(all(b), x) - b.begin());
    return res;
}

// =================== Префиксные суммы ===================
vi prefix_sum(const vi& a) {
    vi p(sz(a) + 1);
    for (int i = 0; i < sz(a); ++i)
        p[i + 1] = p[i] + a[i];
    return p;
}

// =================== Графы ===================
void dfs(int v, vvi& g, vector<bool>& used) {
    used[v] = true;
    for (int u : g[v])
        if (!used[u]) dfs(u, g, used);
}

void bfs(int s, vvi& g, vi& dist) {
    queue<int> q;
    q.push(s);
    dist[s] = 0;
    while (!q.empty()) {
        int v = q.front(); q.pop();
        for (int u : g[v]) {
            if (dist[u] == -1) {
                dist[u] = dist[v] + 1;
                q.push(u);
            }
        }
    }
}

void dijkstra(int s, vvi& g, vi& dist) {
    int n = sz(g);
    dist.assign(n, 1e18);
    dist[s] = 0;
    priority_queue<pii, vector<pii>, greater<>> pq;
    pq.push({0, s});
    while (!pq.empty()) {
        auto [d, v] = pq.top(); pq.pop();
        if (d > dist[v]) continue;
        for (int u : g[v]) {
            if (dist[v] + 1 < dist[u]) {
                dist[u] = dist[v] + 1;
                pq.push({dist[u], u});
            }
        }
    }
}

// =================== Fenwick Tree ===================
struct Fenwick {
    int n;
    vi bit;
    Fenwick(int n) : n(n), bit(n, 0) {}

    void add(int i, int x) {
        for (; i < n; i |= (i + 1))
            bit[i] += x;
    }

    int sum(int r) {
        int res = 0;
        for (; r >= 0; r = (r & (r + 1)) - 1)
            res += bit[r];
        return res;
    }

    int sum(int l, int r) {
        return sum(r) - sum(l - 1);
    }
};

// =================== Segment Tree ===================
struct SegTree {
    int n;
    vi t;
    SegTree(int n) : n(n), t(4 * n, 1e18) {}

    void update(int v, int tl, int tr, int pos, int val) {
        if (tl == tr) t[v] = val;
        else {
            int tm = (tl + tr) / 2;
            if (pos <= tm) update(v * 2, tl, tm, pos, val);
            else update(v * 2 + 1, tm + 1, tr, pos, val);
            t[v] = min(t[v * 2], t[v * 2 + 1]);
        }
    }

    int get_min(int v, int tl, int tr, int l, int r) {
        if (l > r) return 1e18;
        if (l == tl && r == tr) return t[v];
        int tm = (tl + tr) / 2;
        return min(
            get_min(v * 2, tl, tm, l, min(r, tm)),
            get_min(v * 2 + 1, tm + 1, tr, max(l, tm + 1), r)
        );
    }
};

// =================== Подсчёт инверсий ===================
int count_inversions(vi a) {
    int n = sz(a);
    vi b = compress(a);
    Fenwick fw(n);
    int inv = 0;
    for (int i = n - 1; i >= 0; --i) {
        inv += fw.sum(b[i] - 1);
        fw.add(b[i], 1);
    }
    return inv;
}

// =================== Утилиты ===================
template<typename T>
void print_vec(const vector<T>& v) {
    for (auto x : v) cout << x << ' ';
    cout << '\n';
}

// =================== main ===================
int32_t main() {
    fast_io();

    // Пример использования
    int n;
    cin >> n;
    vi a(n);
    for (int& x : a) cin >> x;
    vi psum = prefix_sum(a);
    print_vec(psum);

    return 0;
}

File: template.py
import sys
import math
from bisect import bisect_left
from collections import deque, Counter
from heapq import heappush, heappop
sys.setrecursionlimit(10**6)
input = sys.stdin.readline

# Быстрое возведение в степень
def binpow(a, b, mod):
    res = 1
    a %= mod
    while b:
        if b % 2: res = res * a % mod
        a = a * a % mod
        b //= 2
    return res

# GCD и LCM
def gcd(a, b):
    while b: a, b = b, a % b
    return a

def lcm(a, b):
    return a * b // gcd(a, b)

# Префиксные суммы
def prefix_sum(arr):
    ps = [0]
    for x in arr:
        ps.append(ps[-1] + x)
    return ps

# Бинарный поиск по ответу
def bin_search(ok, ng, check):
    while abs(ok - ng) > 1:
        mid = (ok + ng) // 2
        if check(mid):
            ok = mid
        else:
            ng = mid
    return ok

# DFS — поиск в глубину
def dfs(v, graph, visited):
    visited[v] = True
    for u in graph[v]:
        if not visited[u]:
            dfs(u, graph, visited)

# BFS — поиск в ширину
def bfs(start, graph):
    n = len(graph)
    dist = [-1] * n
    dist[start] = 0
    q = deque([start])
    while q:
        v = q.popleft()
        for u in graph[v]:
            if dist[u] == -1:
                dist[u] = dist[v] + 1
                q.append(u)
    return dist

# Алгоритм Дейкстры — кратчайшие пути в графе с неотрицательными весами
def dijkstra(start, graph, n):
    dist = [float('inf')] * n
    dist[start] = 0
    pq = [(0, start)]
    while pq:
        d, v = heappop(pq)
        if d > dist[v]: continue
        for u, w in graph[v]:
            if dist[v] + w < dist[u]:
                dist[u] = dist[v] + w
                heappush(pq, (dist[u], u))
    return dist

# Z-функция — длины совпадений с префиксом
def z_function(s):
    n = len(s)
    z = [0] * n
    l = r = 0
    for i in range(1, n):
        if i <= r:
            z[i] = min(r - i + 1, z[i - l])
        while i + z[i] < n and s[z[i]] == s[i + z[i]]:
            z[i] += 1
        if i + z[i] - 1 > r:
            l, r = i, i + z[i] - 1
    return z

# Префикс-функция (КМП) — длины префиксов == суффиксам
def prefix_function(s):
    n = len(s)
    pi = [0] * n
    for i in range(1, n):
        j = pi[i - 1]
        while j > 0 and s[i] != s[j]:
            j = pi[j - 1]
        if s[i] == s[j]:
            j += 1
        pi[i] = j
    return pi

# Сжатие координат — уменьшение больших значений до индексов
def compress(arr):
    b = sorted(set(arr))
    return [bisect_left(b, x) for x in arr]

# DSU (система непересекающихся множеств)
class DSU:
    def __init__(self, n):
        self.p = list(range(n))
        self.r = [0] * n

    def find(self, v):
        if v != self.p[v]:
            self.p[v] = self.find(self.p[v])
        return self.p[v]

    def union(self, a, b):
        a, b = self.find(a), self.find(b)
        if a != b:
            if self.r[a] < self.r[b]: a, b = b, a
            self.p[b] = a
            if self.r[a] == self.r[b]: self.r[a] += 1

# Fenwick Tree (Binary Indexed Tree) — суммы на отрезке
class Fenwick:
    def __init__(self, n):
        self.n = n
        self.t = [0] * (n + 1)

    def add(self, i, x):
        i += 1
        while i <= self.n:
            self.t[i] += x
            i += i & -i

    def sum(self, i):
        i += 1
        res = 0
        while i > 0:
            res += self.t[i]
            i -= i & -i
        return res

    def range_sum(self, l, r):
        return self.sum(r) - self.sum(l - 1)

# Segment Tree — минимум/максимум/сумма на отрезке
class SegmentTree:
    def __init__(self, data):
        self.N = len(data)
        self.size = 1
        while self.size < self.N:
            self.size <<= 1
        self.tree = [float('-inf')] * (2 * self.size)
        for i in range(self.N):
            self.tree[self.size + i] = data[i]
        for i in range(self.size - 1, 0, -1):
            self.tree[i] = max(self.tree[2 * i], self.tree[2 * i + 1])

    def update(self, pos, value):
        pos += self.size
        self.tree[pos] = value
        while pos > 1:
            pos >>= 1
            self.tree[pos] = max(self.tree[2 * pos], self.tree[2 * pos + 1])

    def query(self, l, r):
        res = float('-inf')
        l += self.size
        r += self.size
        while l <= r:
            if l % 2 == 1:
                res = max(res, self.tree[l])
                l += 1
            if r % 2 == 0:
                res = max(res, self.tree[r])
                r -= 1
            l //= 2
            r //= 2
        return res
